server.js

```diff
@@ -2257,26 +2259,121 @@ cache(function(data, match, sendBadge, request) {
       return;
     }
     try {
-      var unstable = function(ver) {
-        return /-[0-9A-Za-z.-]+(?:\+[0-9A-Za-z.-]+)?$/.test(ver);
-      };
-      var releases = json[userRepo];
-      if (releases.length == 0) {
-        badgeData.text[1] = 'none';
-        badgeData.colorscheme = 'lightgrey';
-        sendBadge(format, badgeData);
-        return;
+      if (info === 'v') {
+        var unstable = function(ver) {
+          return /-[0-9A-Za-z.-]+(?:\+[0-9A-Za-z.-]+)?$/.test(ver);
+        };
+        var releases = json['releases'];
+        if (releases.length === 0) {
+          badgeData.text[1] = 'none';
+          badgeData.colorscheme = 'lightgrey';
+          sendBadge(format, badgeData);
+          return;
+        }
+        var versions = releases.map(function(version) {
+          return version.version;
+        });
+        var version = latestVersion(versions);
+        if (unstable(version)) {
+          badgeData.colorscheme = "yellow";
+        } else {
+          badgeData.colorscheme = "blue";
+        }
+        badgeData.text[1] = "v" + version;
+      } else if (info === 'dt') {
+        var total = json['downloads'];
+        if (total === 0) {
+          badgeData.colorscheme = 'red';
+        } else if (total < 100) {
+          badgeData.colorscheme = 'yellow';
+        } else if (total < 1000) {
+          badgeData.colorscheme = 'yellowgreen';
+        } else if (total < 10000) {
+          badgeData.colorscheme = 'green';
+        } else {
+          badgeData.colorscheme = 'brightgreen';
```

Comment:
That's a good one. I might want to use it in other PRs too. Is there some kind of generic color function where one could pass the desired values for each color?